### PR TITLE
PB-871 document message module

### PR DESCRIPTION
**Pull request by janpetschexain**
**References**
- [PB-871](https://xainag.atlassian.net/browse/PB-871)

**Summary**
- add module level documentation
- update struct and method documentation
- unravel import indirections

note that this requires `cargo +nightly doc` to generate proper internal links, which is not an issue since this is supported by the hosted rust docs as well.

### CODE REVIEW

File: rust/src/message/payload/update.rs [hunk 1]
```diff
@@ -74,122 +93,97 @@ impl<T: AsRef<[u8]>> UpdateBuffer<T> {
...
         &mut self.inner.as_mut()[offset..]
     }
 }
 
 #[derive(Debug, Eq, PartialEq, Clone)]
-/// High level representation of an update message. These messages are
-/// sent by update partipants during the update phase.
+/// A high level representation of an update message.
+///
+/// These messages are sent by update partipants during the update phase.
```
Comment (Robert-Steiner) [inline rust/src/message/payload/update.rs:170]:
```suggestion
/// These messages are sent by update participants during the update phase.
```

File: rust/src/message/message.rs [hunk 2]
```diff
@@ -1,40 +1,49 @@
...
 }
 
+/// An owned version of a [`Message`].
 pub type MessageOwned = Message<Certificate, LocalSeedDict, MaskObject, MaskObject>;
 
 macro_rules! impl_new {
-    ($name:ident, $payload:ty, $tag:expr) => {
+    ($name:ident, $payload:ty, $tag:expr, $doc:expr) => {
         paste::item! {
+            #[doc = "Creates a new message containing"]
```
Comment (Robert-Steiner) [inline rust/src/message/message.rs:45]: Nice!

File: rust/src/message/mod.rs [hunk 3]
```diff
@@ -1,21 +1,61 @@
...
+//! - The local seed dictionary stores the encrypted mask seed, which generates the local mask for
+//!   the local model, which is encrypted by the ephemeral public keys of the sum participants.
+//!
+//! # The sum2 message
+//! The [`Sum2`] message is an abstraction for the values which a sum participant communicates to
+//! XayNet during the sum2 phase of the PET protocol. It contains the following values:
+//! - The sum signature proves the eligibility of the participant for the sum task.
+//! - The global mask is used by XayNet to unmask the aggregated global model.
+//!
+//! [crypto module]: ../crypto/index.html
```
Comment (Robert-Steiner) [inline rust/src/message/mod.rs:33]: Again, great module documentation👍

File: rust/src/message/traits.rs [hunk 4]
```diff
@@ -135,7 +157,7 @@ impl<T: AsRef<[u8]>> LengthValueBuffer<T> {
         Ok(buffer)
     }
 
-    /// Create a new `LengthValueBuffer` without any bound check.
+    /// Create a new [`LengthValueBuffer`] without any bound check.
```
Comment (Robert-Steiner) [inline rust/src/message/traits.rs:160]:
```suggestion
    /// Create a new [`LengthValueBuffer`] without any bound checks.
```

File: rust/src/message/buffer.rs [hunk 5]
```diff
@@ -102,15 +116,14 @@ impl<T: AsRef<[u8]>> MessageBuffer<T> {
         Ok(buffer)
     }
 
-    /// Return a `MessageBuffer` without performing any bound
-    /// check. This means accessing the various fields may panic if
-    /// the data is invalid.
+    /// Returns a [`MessageBuffer`] without performing any bound check.
```
Comment (Robert-Steiner) [inline rust/src/message/buffer.rs:119]:
```suggestion
    /// Returns a [`MessageBuffer`] without performing any bound checks.
```

File: rust/src/message/buffer.rs [hunk 6]
```diff
@@ -1,12 +1,22 @@
...
 use crate::{
-    message::{utils::range, DecodeError, Flags, LengthValueBuffer},
+    message::{header::Flags, traits::LengthValueBuffer, utils::range, DecodeError},
     CoordinatorPublicKey,
     ParticipantPublicKey,
 };
 
+/// Gets the length of the message header in bytes.
+///
+/// Depends on the inluded certificates which can be variable in length.
```
[the remark this answers is not in the document]
Comment (janpetschexain) [reply rust/src/message/buffer.rs:19]: changed.

File: rust/src/message/payload/update.rs (hunk 1 above)
Comment (janpetschexain) [reply rust/src/message/payload/update.rs:170]: changed.

File: rust/src/message/traits.rs (hunk 4 above)
Comment (janpetschexain) [reply rust/src/message/traits.rs:160]: changed.

File: rust/src/message/buffer.rs (hunk 5 above)
Comment (janpetschexain) [reply rust/src/message/buffer.rs:119]: changed.

**Pull request merged**